src/main/webapp/app/shared/monaco-editor/model/actions/emoji.action.ts

```diff
@@ -0,0 +1,121 @@
+import { TextEditorAction } from 'app/shared/monaco-editor/model/actions/text-editor-action.model';
+import { faSmile } from '@fortawesome/free-solid-svg-icons';
+import { TextEditor } from 'app/shared/monaco-editor/model/actions/adapter/text-editor.interface';
+import { ViewContainerRef } from '@angular/core';
+import { EmojiPickerComponent } from 'app/shared/metis/emoji/emoji-picker.component';
+import { Overlay, OverlayPositionBuilder, OverlayRef } from '@angular/cdk/overlay';
+import { ComponentPortal } from '@angular/cdk/portal';
+import { TextEditorPosition } from 'app/shared/monaco-editor/model/actions/adapter/text-editor-position.model';
+import { TextEditorRange } from 'app/shared/monaco-editor/model/actions/adapter/text-editor-range.model';
+
+/**
+ * Action to open the emoji picker and insert the selected emoji into the editor.
+ */
+export class EmojiAction extends TextEditorAction {
+    static readonly ID = 'emoji.action';
+    private overlayRef: OverlayRef | null = null;
+    private position?: { x: number; y: number };
+
+    constructor(
+        private viewContainerRef: ViewContainerRef,
+        private overlay: Overlay,
+        private positionBuilder: OverlayPositionBuilder,
+    ) {
+        super(EmojiAction.ID, 'artemisApp.multipleChoiceQuestion.editor.emoji', faSmile, undefined);
```

Comment:
Maybe here the new injection technique works too, you'll probably still need the constructor for the super call though. 
I also agree with code rabbit here, that it's confusing that `multipleChoiceQuestion` is used.
I don't think it makes sense to put the translation into these files

Reply:
I followed this approach for the translation because it was the same in all the other action files. I think this could be refactored in a follow-up PR for consistency across all action files.